src/Desktop/ArcGISRuntime.Desktop.Samples/Samples/Map/AuthorMap/AuthorMap.xaml.cs

```diff
@@ -0,0 +1,497 @@
+            Envelope currentExtent = currentViewpoint.TargetGeometry as Envelope;
+            // Project the current extent to geographic coordinates (longitude / latitude)
+            Envelope currentGeoExtent = GeometryEngine.Project(currentExtent, SpatialReferences.Wgs84) as Envelope;
+
+            // Fill the app text boxes with min / max longitude (x) and latitude (y) to four decimal places
+            XMinTextBox.Text = currentGeoExtent.XMin.ToString("0.####");
+            YMinTextBox.Text = currentGeoExtent.YMin.ToString("0.####");
+            XMaxTextBox.Text = currentGeoExtent.XMax.ToString("0.####");
+            YMaxTextBox.Text = currentGeoExtent.YMax.ToString("0.####");
+        }
+
+        #region OAuth helpers
+        private void UpdateAuthenticationManager()
+        {
+            // Register the server information with the AuthenticationManager
+            Esri.ArcGISRuntime.Security.ServerInfo portalServerInfo = new ServerInfo
+            {
+                ServerUri = new Uri(ServerUrl),
+                OAuthClientInfo = new OAuthClientInfo
+                {
+                    ClientId = AppClientId,
+                    RedirectUri = new Uri(OAuthRedirectUrl)
+                },
+                // Specify OAuthAuthorizationCode if you need a refresh token (and have specified a valid client secret)
+                // Otherwise, use OAuthImplicit
+                TokenAuthenticationType = TokenAuthenticationType.OAuthImplicit
+            };
+            AuthenticationManager.Current.RegisterServer(portalServerInfo);
+
+            // Use the OAuthAuthorize class in this project to create a new web view that contains the OAuth challenge handler.
+            AuthenticationManager.Current.OAuthAuthorizeHandler = new OAuthAuthorize();
+
+            // Create a new ChallengeHandler that uses a method in this class to challenge for credentials
+            AuthenticationManager.Current.ChallengeHandler = new Esri.ArcGISRuntime.Security.ChallengeHandler(CreateCredentialAsync);
+        }
+
+        private async Task<bool> EnsureLoginToArcGISAsync()
+        {
+            var authenticated = false;
+
+            // Create an OAuth credential request for arcgis.com
+            CredentialRequestInfo loginInfo = new CredentialRequestInfo();
+
+            // Use the OAuth implicit grant flow
+            loginInfo.GenerateTokenOptions = new GenerateTokenOptions
+            {
+                TokenAuthenticationType = TokenAuthenticationType.OAuthImplicit
+            };
+
+            // Indicate the url (portal) to authenticate with (ArcGIS Online)
+            loginInfo.ServiceUri = new Uri("http://www.arcgis.com/sharing/rest");
+
+            try
+            {
+                // Call GetCredentialAsync on the AuthenticationManager to invoke the challenge handler
+                var cred = await AuthenticationManager.Current.GetCredentialAsync(loginInfo, false);
```

Comment:
Consider changing to the following to avoid the run-on sentence type of coding:

                // Call GetCredentialAsync on the AuthenticationManager to invoke the challenge handler
                AuthenticationManager myAuthenticationManager = AuthenticationManager.Current
                var cred = await myAuthenticationManager.GetCredentialAsync(loginInfo, false);
                authenticated = (cred != null);